### ISSUE TITLE
SpirvCapabilityNotSupported SampledCubeArray

### ISSUE THREAD

**Issue by gregjohnson2017** [vulkano-rs/vulkano#1981]
# Template

* Version of vulkano: Tested on release 0.30.0 and master
* OS: Windows 10
* GPU (the selected PhysicalDevice): NVIDIA GeForce GTX 980 Ti (type: DiscreteGpu)
* GPU Driver: GeForce Game Ready Driver version 516.94

# Issue

I'm trying to use a samplerCubeArray in a GLSL fragment shader and whenever I load a basic shader, I get an error saying that it doesn't support the capability.

```rust
mod fs {
    vulkano_shaders::shader! {
        ty: "fragment",
        src: "
            #version 450

            layout(location = 0) in vec3 v_color;
            layout(location = 0) out vec4 f_color;
            layout(location = 0, binding = 0) uniform samplerCubeArray cubeMapArray;

            void main() {
                vec4 fullBright = texture(cubeMapArray, vec4(1.0,0.0,0.0, 0));
                f_color = vec4(v_color, 1.0) + fullBright;
            }
        "
    }
}

let fs = fs::load(device.clone()).unwrap(); // <--- runtime error here
```
```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: SpirvCapabilityNotSupported { capability: SampledCubeArray, reason: RequirementsNotMet(["feature `image_cube_array`"]) }', src\graphics.rs:136:43
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
error: process didn't exit successfully: `target\debug\rtvox.exe` (exit code: 101)
```

**Comment by Rua** [comment]
Have you enabled the device feature that is mentioned in the error message?

**Comment by gregjohnson2017** [comment]
What device feature would that be? All of the internet searching I've done has indicated that this is something that is included by default and not something you can explicitly enable.

**Comment by Rua** [comment]
It's this feature:
https://docs.rs/vulkano/latest/vulkano/device/struct.Features.html#structfield.image_cube_array

**Comment by gregjohnson2017** [comment]
Thanks, you're our hero! I had instance extensions and device extensions from the triangle example for the initial setup of my project, but I had no idea that device features were something else that could have been enabled. Now I see the section in the vulkan spec about VkPhysicalDeviceFeatures. I think what I did wrong last time is I searched the whole spec for "cube_array" and I got a bunch of results, and didn't see anything about device features. I guess I should have searched for the camel case also.
